plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -145,6 +145,141 @@ func NoDiskConflict(pod *api.Pod, existingPods []*api.Pod, node string) (bool, e
 	return true, nil
 }
 
+type MaxPDVolumeCountChecker struct {
+	filter     PDVolumeFilter
+	maxVolumes uint
+	pvInfo     PersistentVolumeInfo
+	pvcInfo    PersistentVolumeClaimInfo
+}
+
+// Information for how to filter PD Volumes when checking PD Volume caps
+type PDVolumeFilter struct {
+	// Filter normal volumes
+	FilterVolume           func(vol *api.Volume) (id string, relevant bool)
+	FilterPersistentVolume func(pv *api.PersistentVolume) (id string, relevant bool)
+}
+
+// MaxPDVolumeCountPredicate evaluates whether a pod can fit based on the number of volumes which match
```

Comment:
I was going based on the style of `NewVolumeZonePredicate` (which puts the predicate documentation on the `New` function), but there doesn't quite seem to be a consistent style in this file (other predicates put the predicate documentation and the predicate function, and don't have any documentation on the new function, etc).  What is the desired style here (I'm somewhat loathe to introduce yet another style here).